src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseExceptionThrowHelpers.cs

```diff
@@ -282,6 +290,19 @@ public override void Initialize(AnalysisContext context)
                     }
                 }, OperationKind.Throw);
 
+                // As a heuristic, we only want to replace throws with ThrowIfNull if either there isn't currently
+                // a specified parameter name, e.g. the parameterless constructor was used, or if it's specified in a
+                // simple fashion, e.g. a nameof or a literal string.  This is primarily to avoid false positives
+                // with complicated expressions for computing the parameter name to use, which with ThrowIfNull would
+                // need to be done prior to the guard check, and thus something we want to avoid.
+                bool HasReplaceableArgumentName(IObjectCreationOperation creationOperation, int argumentIndex)
```

Comment:
Relying on `argumentIndex` might fail for named arguments if they are out of parameter order

Reply:
We have a helper method for getting the argument operation corresponding to a given parameter index: https://github.com/dotnet/roslyn-analyzers/blob/b7bb138809d5a7d31508fe0cd86d59ed4c864764/src/Utilities/Compiler/Extensions/IOperationExtensions.cs#L850-L852